core/trino-main/src/main/java/io/trino/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -454,24 +500,26 @@ public void addInput(GroupByIdBlock groupIdsBlock, Page page)
         @Override
         public void addIntermediate(GroupByIdBlock groupIdsBlock, Block block)
         {
-            throw new UnsupportedOperationException();
+            spillableStateHolder.orElseThrow(UnsupportedOperationException::new).addIntermediate(groupIdsBlock, block);
         }
 
         @Override
         public void evaluateIntermediate(int groupId, BlockBuilder output)
         {
-            throw new UnsupportedOperationException();
+            spillableStateHolder.orElseThrow(UnsupportedOperationException::new).evaluateIntermediate(groupId, output);
         }
 
         @Override
         public void evaluateFinal(int groupId, BlockBuilder output)
         {
+            spillableStateHolder.ifPresent(holder -> holder.verifyEvaluateFinal());
             accumulator.evaluateFinal(groupId, output);
         }
 
         @Override
         public void prepareFinal()
         {
+            spillableStateHolder.ifPresent(holder -> holder.prepareFinal((groupIdBlock, page) -> addInputToUnderlyingAccumulator(groupIdBlock, page)));
```

Comment:
nit: it's a little difficult to read, I wonder if it would be better to rewrite it without using lambdas?  (same for `OrderingGroupedAccumulator`)

Reply:
Done